trace/run.go

```diff
@@ -8,20 +8,27 @@ import (
 	akihttp "github.com/akitasoftware/akita-libs/akinet/http"
 )
 
-func Collect(stop <-chan struct{}, intf, bpfFilter string, proc Collector) error {
+func Collect(stop <-chan struct{}, intf, bpfFilter string, proc Collector, packetCount PacketCountConsumer) error {
 	defer proc.Close()
 
 	facts := []akinet.TCPParserFactory{
 		akihttp.NewHTTPRequestParserFactory(),
 		akihttp.NewHTTPResponseParserFactory(),
 	}
+
 	parser := col.NewNetworkTrafficParser()
+
+	if packetCount != nil {
+		parser.InstallObserver(CountTcpPackets(intf, packetCount))
+	}
+
 	parsedChan, err := parser.ParseFromInterface(intf, bpfFilter, stop, facts...)
 	if err != nil {
 		return errors.Wrap(err, "couldn't start parsing from interface")
 	}
 
 	for t := range parsedChan {
+		t.Interface = intf
```

Comment:
Why not fold this into `ParseFromInterface`?

Reply:
That makes a lot of sense, but I could not make it work easily.  Some ParsedNetworkPackets are generated directly in parser, but the ones we care about come from TCP reassembly, so the interface name would have to be added to that context so that the callback which eventually creates the HTTP Response could find it.

I think the appropriate place would be 
```
type assemblerCtxWithSeq struct {
	ci       gopacket.CaptureInfo
	seq, ack reassembly.Sequence
}
```

Reply:
Maybe ParseFromInterface could have a separate goroutine to re-stamp every ParsedNetworkPacket with the right interface, and put it on a different channel?  Still a bunch of work to get right.